experimental/sql/src/impl_to_iterators.py

```diff
@@ -258,6 +258,21 @@ def match_and_rewrite(self, op: RelImpl.FullTableScanOp,
                                  convert_bag(op.result.typ)))
 
 
+@dataclass
+class PartialTableScanRewriter(RelImplRewriter):
+
+  table_mapping: dict[str, it.ColumnarBatch]
+
+  @op_type_rewrite_pattern
+  def match_and_rewrite(self, op: RelImpl.PartialTableScanOp,
+                        rewriter: PatternRewriter):
+    rewriter.replace_matched_op(
+        it.ScanColumnarBatch.get(
```

Comment:
I suppose that this assumes that the table mapping has been populated with column batch arguments that have the corresponding projection applied, right? Just to be sure: the logic that applies these projections isn't part of this commit, yet, right? Can you explain this mechanism in comments?

Reply:
Oh, I see, `get_batch_and_name_list` applies the projections. Can you explain that here, where get_batch_and_name_list` does the projection, and in the documentation of that function?

Reply:
Actually, it's not really get_batch_and_name_list that does the projection since we want to project on to SSAValues (in this case BlockArguments matching the right table). And these block arguments can only be built later on.

So this function really only returns batchtypes and names and does not do any mapping.

I will add some comments to make this clearer.

Reply:
Aha, right, the actual projection is done even earlier, completely outside of the plan. `get_batch_and_name_list` populates the column batches that are accessed by some scan op, some of which may be projected before the call into the plan. The new comments are better, but it doesn't mention yet that there is this convention between the call site and the plan. Can you elaborate on that?

Reply:
Are you referring to the projection of the columnar batch itself? I am not sure whether this is actually the right approach. We might have to think about changing your scan_columnar_batch at some point to also take col names and only builds tuples from these columns.

I changed the comment again :)

Reply:
You haven't pushed that comment yet, have you?

Reply:
Yes, indeed. It's there now :)

Reply:
Thanks for the update! I still don't understand where exactly the projection happens, though. As I understand it, the lowered code can use some SSA values with tables where the projection has already been applied (i.e., a columnar batch with only a subset of the original columns) and the table mapping contains the names of those SSA values. But who creates these values?